lib/puppet/network/http/api/indirected_routes.rb

```diff
@@ -111,24 +107,6 @@ def uri2indirection(http_method, uri, params)
 
   private
 
-  def do_http_control_exception(response, exception)
-    msg = exception.message
-    Puppet.info(msg)
-    response.respond_with(exception.status, "text/plain", msg)
-  end
-
-  def do_exception(response, exception, status=400)
-    if exception.is_a?(Puppet::Network::AuthorizationError)
-      # make sure we return the correct status code
-      # for authorization issues
-      status = 403 if status == 400
-    end
```

Comment:
Ahh, good catch. In my head when reading it I substituted `Puppet::Network::HTTP::Error::HTTPNotAuthorizedError` (which will behave as expected with the error handling at the higher level).

Implementation-wise, since we have an [v3Auth class](https://github.com/puppetlabs/puppet/blob/98d22be27447bcb34c33aa4a44610ab8768390f5/lib/puppet/network/http/api/master/v3/authorization.rb) that's calling the auth provider and wrapping any `AuthorizationError`s in the appropriate HTTPError for our environment(s) [routes](https://github.com/puppetlabs/puppet/blob/98d22be27447bcb34c33aa4a44610ab8768390f5/lib/puppet/network/http/api/master/v3.rb#L13-L16). But, IIUC, we need to do the URI munging in `uri2indirection` before we can call our authorization for indirected routes. I would, at this point in my thinking, wrap [the call to `check_authorization`](https://github.com/puppetlabs/puppet/blob/98d22be27447bcb34c33aa4a44610ab8768390f5/lib/puppet/network/http/api/indirected_routes.rb#L95) in a similar catch and wrap block to what the v3Auth class is doing. That goes with the generic HTTPHandler only really caring about proper HTTPErrors and that we're already doing a lot of check and raise work in `uri2indirection` (but I don't know if the fact we're already doing a bunch of catch-and-raise work necessarily makes it a good thing to add more...).

Does that approach sound sane, or do you think that handling general Auth errors is something the Handler should know about, or is there better route that I haven't investigated (eg. I could see wanting to refactor `uri2indirection` into a set of smaller methods but I don't know if it's something that would be valuable within the scope of this work).